diff --git a/qtbase/src/corelib/tools/qsimd_p.h b/qtbase/src/corelib/tools/qsimd_p.h
--- a/qtbase/src/corelib/tools/qsimd_p.h
+++ b/qtbase/src/corelib/tools/qsimd_p.h
@@ -466,5 +466,13 @@
 #define qCpuHasFeature(feature)     ((qCompilerCpuFeatures & (Q_UINT64_C(1) << CpuFeature ## feature)) \
                                      || (qCpuFeatures() & (Q_UINT64_C(1) << CpuFeature ## feature)))
+#if defined(Q_CC_CLANG)
+#ifdef _bit_scan_reverse
+#  undef _bit_scan_reverse
+#endif
+#ifdef _bit_scan_forward
+#  undef _bit_scan_forward
+#endif
+#endif
 #if QT_HAS_BUILTIN(__builtin_clz) && QT_HAS_BUILTIN(__builtin_ctz) && defined(Q_CC_CLANG) && !defined(Q_CC_INTEL)
 static Q_ALWAYS_INLINE unsigned _bit_scan_reverse(unsigned val)
 {
